### PR TITLE
(php) Extending php package functionality

### PR DESCRIPTION
**Pull request by AdmiringWorm**
### Changes

- [X] Allow user to decide between thread safe and non thread safe version (https://github.com/chocolatey/chocolatey-coreteampackages/issues/510#issuecomment-275174820)
- [x] Install to versioned install directory (https://github.com/ferventcoder/chocolatey-packages/issues/115)
- [x] Move configuration file when directory changes
- [x] Allow user to override install directory (*undecided*)
- [x] PHP doesn't respect proxy calls (*Possible to use `Get-WebHeaders` for this?*) (https://github.com/ferventcoder/chocolatey-packages/issues/38)
- [x] Allows users to opt out of added install directory to path

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
@chocolatey/chocolateycommunitypackages anyone up for reviewing this PR?

File: automatic/php/php.nuspec [hunk 1]
```diff
@@ -8,7 +8,16 @@
     <authors>PHP Authors</authors>
     <owners>chocolatey, Rob Reynolds</owners>
     <summary>PHP – widely-used general-purpose scripting language that is especially suited for Web development and can be embedded into HTML.</summary>
-    <description>PHP is an HTML-embedded scripting language. Much of its syntax is borrowed from C, Java and Perl with a couple of unique PHP-specific features thrown in. The goal of the language is to allow web developers to write dynamically generated pages quickly.</description>
+    <description>PHP is an HTML-embedded scripting language. Much of its syntax is borrowed from C, Java and Perl with a couple of unique PHP-specific features thrown in. The goal of the language is to allow web developers to write dynamically generated pages quickly.
+
+## Package Parameters
+- `/DontAddToPath` - Do not add install directory to path
+- `/InstallDir:`   - Override the installation directory (needs to be specified both during install and update, until it is remembered by choco)
```
Comment (ferventcoder) [inline automatic/php/php.nuspec]: maybe - the underlying installer may remember where it is installed for upgrades

Comment (AdmiringWorm) [inline automatic/php/php.nuspec]: this package doesn't use any installer, it uses the provided zip archives.

Comment (ferventcoder) [inline automatic/php/php.nuspec]: oh right on!

**Comment by photodude** [comment]
@AdmiringWorm Is there a guide somewhere that would explain how to use the side-by-side switch `install -m` for Installing to a versioned install directory? I'm trying to understand how to implement that on AppVeyor for caching different installs like php 7 vs PHP 7.1.

**Comment by AdmiringWorm** [comment]
@photodude 
Do allow side by side installation I think there is enough to just pass the following:
`choco install -m php --version {version} --params '"/InstallDir:{Path}"'`
it is possible that you may have to use the `--force` argument as well, but not sure about that.

Do note though, the changes to add support for side-by-side installation have not yet been made available for the following versions: 7.1.x, 7.0.x and 5.6.x.

**Pull request merged**